go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -324,6 +325,180 @@ func (oc *Controller) deleteEgressFirewallRules(externalID string) error {
+	for _, logicalSwitch := range logicalSwitches {
+		for _, match := range matchConditions {
+			_, stderr, err := util.RunOVNNbctl(
+				"acl-del",
+				logicalSwitch,
+				types.DirectionFromLPort,
+				types.EgressFirewallDefaultInternalClusterPriority,
+				match,
+			)
+			if err != nil {
+				klog.Errorf("Unable to delete default allow ACL on switch: %s, stderr: %s, err: %v", logicalSwitch, stderr, err)
+			}
+		}
+	}
+}
+
+// deleteDefaultAllowACLsForEgressFirewall deletes the default allow ACLs as follows:
+// - allow cluster subnet on every nodes' switch
+// - allows service subnet on every nodes' switch
+// - all allow node IPs on every nodes' switch
+func (oc *Controller) deleteDefaultAllowACLsForEgressFirewall() {
+	matchConditions := []string{}
+	v4ClusterSubnet, v6ClusterSubnet := getClusterSubnets()
+	v4ServiceSubnet, v6ServiceSubnet := getServiceSubnets()
+	if config.Gateway.Mode == config.GatewayModeLocal {
+		matchConditions = append(matchConditions, generateDefaultClusterSubnetAllowMatch(v4ClusterSubnet, v6ClusterSubnet)...)
+		matchConditions = append(matchConditions, generateDefaultServiceSubnetAllowMatch(v4ServiceSubnet, v6ServiceSubnet, v4ClusterSubnet, v6ClusterSubnet)...)
+	}
+	nodes, err := oc.watchFactory.GetNodes()
+	if err != nil {
+		klog.Errorf("Unable to delete default allow node ACLs, error fetching nodes, err: %v", err)
+	}
+	for _, node := range nodes {
+		matchConditions = append(matchConditions, oc.generateDefaultNodeAllowMatch(node, v4ClusterSubnet, v6ClusterSubnet)...)
+	}
+	logicalSwitches := oc.getEgressFirewallACLSwitches()
+	for _, logicalSwitch := range logicalSwitches {
+		for _, match := range matchConditions {
+			_, stderr, err := util.RunOVNNbctl(
+				"acl-del",
+				logicalSwitch,
+				types.DirectionFromLPort,
+				types.EgressFirewallDefaultInternalClusterPriority,
+				match,
+			)
+			if err != nil {
+				klog.Errorf("Unable to delete default allow ACL on switch: %s, stderr: %s, err: %v", logicalSwitch, stderr, err)
+			}
+		}
+	}
+}
+
+func generateDefaultClusterSubnetAllowMatch(v4ClusterSubnets, v6ClusterSubnets []*net.IPNet) []string {
+	matchConditions := []string{}
+	for _, v4Subnet := range v4ClusterSubnets {
+		matchConditions = append(matchConditions, fmt.Sprintf("ip4.src == %s && ip4.dst == %s", v4Subnet.String(), v4Subnet.String()))
+	}
+	for _, v6Subnet := range v6ClusterSubnets {
+		matchConditions = append(matchConditions, fmt.Sprintf("ip6.src == %s && ip6.dst == %s", v6Subnet.String(), v6Subnet.String()))
+	}
+	return matchConditions
+}
+
+func generateDefaultServiceSubnetAllowMatch(v4ServiceSubnet, v6ServiceSubnet *net.IPNet, v4ClusterSubnets, v6ClusterSubnets []*net.IPNet) []string {
+	matchConditions := []string{}
+	if v4ServiceSubnet != nil {
+		for _, v4Subnet := range v4ClusterSubnets {
+			matchConditions = append(matchConditions, fmt.Sprintf("ip4.src == %s && ip4.dst == %s", v4Subnet.String(), v4ServiceSubnet.String()))
```

Comment:
in these functions, why do you care about matching source? Shouldn't just dest be enough?

Reply:
Because I want to make sure it only matches packets for which the source IP is a pod IP.

Reply:
What other traffic would be there? mp0 would be snat'ed to the pod subnet anyway, so that one is out. Is it to avoid matching on from external to pod service/subnet as a response? What about incoming traffic from outside the cluster that would be snat'ed to join switch then forwarded into the switch towards a pod? I need to think about these some more...

Reply:
Hmm, good question, I overlooked the fact that I am doing this on the GR. Let me run some tests and double check the outcome of removing the source IP from the filter.

Reply:
We can remove the source IP. My tests show that it's fine, and logically given what has been stated here it should also be fine.

Reply:
@alexanderConstantinescu meant ovn worker switch

I'm wondering if we need allow-related on these ACLs. Let's take an example where a pod is an endpoint to a service.
external traffic -> GR LB -> pod, and lets say there is an egress firewall on the pod

The allow will allow the traffic to reach the pod, but then the pod response will be pod -> external IP and will get dropped. Shouldn't it have allow-related to allow it to send the response?

@dceara do you mind reviewing this. ACLs are tricky and can have some implications. I'd appreciate your input here.

Reply:
@alexanderConstantinescu what is the resolution here? are we no longer optimizing ACLs? Removing inverse matching? Maybe we need to have a sync up meeting on this.

Reply:
> are we no longer optimizing ACLs? 

Yes, I removed all the commits related to that.